docs/site/Inside-Loopback-Application.md

```diff
@@ -6,81 +6,146 @@ sidebar: lb4_sidebar
 
-## Starting with the API endpoints
+> Models can also be generated from [existing tables](https://loopback.io/doc/en/lb4/Discovering-models.html) in relational databases.
 
+[User](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/models/user.model.ts) and [Order](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/models/order.model.ts) are domain objects in the shopping cart use case:
+  - They have the same structure and relations when posted by users from the front end as well as when persisted towards a backend datastore.
+  - A user can have multiple orders, and hence `User` model has a [has-many relation](https://loopback.io/doc/en/lb4/HasMany-relation.html) (ie., one-to-many) with `Order` model. Hence the `orders` property in `User` model is annotated with the `@hasMany(() => Order)` decorator.
+  - An `Order` is composed with a list of products added in the shopping cart and submitted towards the order. Hence the `products` property in `Order` model is annotated as `@property.array()` to indicate that it is a list property.
+  - An `Order` can be made only by one `User`. So, the `userId` property in `Order` model is annotated with `@belongsTo()` decorator, to indicate `Order` has a one-to-one relation with `User`.
 
-### configuring the controllers
+## Entity layer and Database connectivity
 
+Applications displaying a `user` profile might need a list of associated recent `orders` made by the user. The same also applies for an order page, some associated user data could be needed. Querying for associated data for demands from front end as well as for logical and persistence reasons is a common requirement. Having a separate `Entity` layer helps in model driven CRUD operations.
 
-### creating the models
+> [Repositories](https://loopback.io/doc/en/lb4/Repositories.html) in LoopBack represent the `Entity` layer for querying and persisting a domain model. They also connect with other repositories to resolve `Entity relations`.
 
+From the exampple,
+  - To query and persist `User` model data, the example uses a [UserRepository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/user.repository.ts).
+  - For the `Order` model, [OrderRepository](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/repositories/order.repository.ts) is defined.
+  - The `UserRepository` uses the entity composition defined in the `User` model to create a has-many relation with the `OrderRepository` 
 
+Now that we have the most of the API implementation next step is to choose appropriate drivers to connect with the back end datasoruces.
```

Comment:
I know we are introducing a lot of concepts for the user in this document.

Perhaps we should change the wording here with regards to

> next step is to choose appropriate drivers to connect with the back end datasoruces.

In our `gettings started` examples, choosing a datasource comes after creating a model and before repositories and controllers.

e.g. https://loopback.io/doc/en/lb4/todo-tutorial.html 

Perhaps simply mention that repos communicate with backend datasources....and don't mention that we are following a certain series of steps.

And change 

> datasoruces. 

to 

datasources

Reply:
agreed